### PR TITLE
RUN-4443 - Channel Lifecycle / onDisconnection

### PR DESCRIPTION
**Pull request by tgoc99**
Adding onDisconnection and cleaning up channel.connections on a disconnect.

[new testrunner](https://testing-dashboard.openfin.co/#/app/tests/5b56102154b21953031f37a7/edit)

[Win7](https://testing-dashboard.openfin.co/#/app/sessions/api/completed/5bb3e9cecb360141a7dfcffe)
[Win10](https://testing-dashboard.openfin.co/#/app/sessions/api/completed/5bb3eafdcb360141a7dfd000)

### CODE REVIEW

File: src/api/interappbus/channel/index.ts [hunk 1]
```diff
@@ -95,6 +95,14 @@ export class Channel extends EmitterBase<ChannelEvents> {
         const channel = new ChannelProvider(providerIdentity, this.wire.sendAction.bind(this.wire));
         const key = providerIdentity.channelId;
         this.channelMap.set(key, channel);
+        this.on('client-disconnected', (eventPayload: ProviderIdentity) => {
```
Comment (rdepena) [inline src/api/interappbus/channel/index.ts]: Is this raised on window close from the core as well ?

File: src/api/interappbus/channel/provider.ts [hunk 2]
```diff
@@ -5,8 +5,8 @@ import { Identity } from '../../../main';
 export type ConnectionListener = (identity: Identity, connectionMessage?: any) => any;
 
 export class ChannelProvider extends ChannelBase {
-    private connectListener: ConnectionListener;
```
Comment (pbaize) [inline src/api/interappbus/channel/provider.ts]: Do we want to make these public?

File: src/api/interappbus/channel/index.ts (hunk 1 above)
Comment (tgoc99) [reply src/api/interappbus/channel/index.ts]: it uses the subscription manager so should work on window or frame close and external connection close

Comment (tgoc99) [reply src/api/interappbus/channel/index.ts]: i.e. client-disconnected is raised on those other events

File: src/api/interappbus/channel/provider.ts (hunk 2 above)
Comment (pbaize) [reply src/api/interappbus/channel/provider.ts]: Would it be better to add a method that handles this and keep these private?

**Pull request merged**